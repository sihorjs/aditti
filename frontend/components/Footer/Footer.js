import React from "react";
import PropTypes from "prop-types";
import "./Footer.less";

function Footer(props) {
    return (
        <footer className="footer">
            {props.linksList.length && (
                <div className="footer__links-wrapper wrapper">
                    {props.linksList.map(linksList => {
                        return (
                            <div className="footer__links" key={linksList.name}>
                                <div className="footer__links-title">
                                    {linksList.name}
                                </div>
                                <ul className="footer__links-list">
                                    {linksList.links.map(item => {
                                        return (
                                            <li key={item.name}>
                                                <a
                                                    className="footer__link"
                                                    href={item.href}
                                                >
                                                    {item.name}
                                                </a>
                                            </li>
                                        );
                                    })}
                                </ul>
                            </div>
                        );
                    })}
                </div>
            )}
            <div className="footer__copyright">Copyright 2013 CSS Author</div>
        </footer>
    );
}

Footer.propTypes = {
    linksList: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            links: PropTypes.arrayOf(
                PropTypes.shape({
                    name: PropTypes.string.isRequired,
                    href: PropTypes.string.isRequired
                }).isRequired
            ).isRequired
        }).isRequired
    )
};

export default Footer;
