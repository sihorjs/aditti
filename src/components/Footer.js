import React from "react";
import PropTypes from "prop-types";

function Footer(props) {
    return (
        <footer>
            <div className="wrapper">
                {props.linksList.map((obj, index) => {
                    return (
                        <div className="links" key={index}>
                            <h3>{obj.name}</h3>
                            <ul>
                                {obj.links.map((item, index) => {
                                    return (
                                        <li key={index}>
                                            <a href={item.href}>{item.name}</a>
                                        </li>
                                    );
                                })}
                            </ul>
                        </div>
                    );
                })}
            </div>
            <div className="copyright">
                <small>Copyright 2013 CSS Author</small>
            </div>
        </footer>
    );
}

Footer.propTypes = {
    linksList: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            links: PropTypes.arrayOf(
                PropTypes.shape({
                    name: PropTypes.string.isRequired,
                    href: PropTypes.string.isRequired
                }).isRequired
            ).isRequired
        }).isRequired
    )
};

export default Footer;
