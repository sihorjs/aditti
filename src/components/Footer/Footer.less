@import url("../../less/index.less");

.footer {
    background-color: @cyan;
    color: @white;

    &__links-wrapper {
        display: flex;
        justify-content: space-between;
        width: @desktop-width;
        margin: 0 auto;
        padding: 30px 0;
    }

    &__links-title {
        margin-bottom: 15px;
        font: 27px @heading-font;
        text-transform: uppercase;
    }

    &__link {
        font: 15px/28px @main-font;
        color: @white;

        &:hover {
            text-decoration: underline;
        }
    }

    &__copyright {
        display: flex;
        justify-content: center;
        align-items: center;
        height: 40px;
        font: 15px @main-font;
        background-color: @darkcyan;
    }
}
