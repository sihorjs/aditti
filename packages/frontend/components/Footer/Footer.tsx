import React from "react";
import FooterLinksType from "@aditti/types/footer";
import "./Footer.less";

export type Props = {
    linksList: FooterLinksType;
};

function Footer(props: Props) {
    return (
        <footer className="footer">
            {props.linksList.length && (
                <div className="footer__links-wrapper wrapper">
                    {props.linksList.map(linksList => {
                        return (
                            <div className="footer__links" key={linksList.name}>
                                <div className="footer__links-title">
                                    {linksList.name}
                                </div>
                                <ul className="footer__links-list">
                                    {linksList.links.map(item => {
                                        return (
                                            <li key={item.name}>
                                                <a
                                                    className="footer__link"
                                                    href={item.href}
                                                >
                                                    {item.name}
                                                </a>
                                            </li>
                                        );
                                    })}
                                </ul>
                            </div>
                        );
                    })}
                </div>
            )}
            <div className="footer__copyright">Copyright 2013 CSS Author</div>
        </footer>
    );
}

export default Footer;
