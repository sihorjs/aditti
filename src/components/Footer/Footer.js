import React from "react";
import PropTypes from "prop-types";
import "./Footer.less";

function Footer(props) {
    return (
        <footer className="footer">
            <div className="footer__links-wrapper">
                {props.linksList.map((obj, index) => {
                    return (
                        <div className="footer__links" key={index}>
                            <div className="footer__links-title">
                                {obj.name}
                            </div>
                            <ul className="footer__links-list">
                                {obj.links.map((item, index) => {
                                    return (
                                        <li key={index}>
                                            <a
                                                className="footer__link"
                                                href={item.href}
                                            >
                                                {item.name}
                                            </a>
                                        </li>
                                    );
                                })}
                            </ul>
                        </div>
                    );
                })}
            </div>
            <div className="footer__copyright">Copyright 2013 CSS Author</div>
        </footer>
    );
}

Footer.propTypes = {
    linksList: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            links: PropTypes.arrayOf(
                PropTypes.shape({
                    name: PropTypes.string.isRequired,
                    href: PropTypes.string.isRequired
                }).isRequired
            ).isRequired
        }).isRequired
    )
};

export default Footer;
