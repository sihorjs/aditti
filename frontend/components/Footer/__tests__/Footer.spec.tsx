import React from "react";
import { shallow } from "enzyme";
import Footer from "../Footer";
import FooterLinksType from "types/footer";

describe("<Footer />", () => {
    const linksList: FooterLinksType = [
        {
            name: "featured sale",
            links: [
                { name: "Alexis Hudson", href: "#" },
                { name: "American Apparel", href: "#" },
                { name: "Ben Sherman", href: "#" }
            ]
        }
    ];

    it("renders without crash", () => {
        const tree = shallow(<Footer linksList={linksList} />);

        expect(tree).toMatchSnapshot();
    });

    it("has .footer class of wrapper", () => {
        const component = shallow(<Footer linksList={linksList} />);

        expect(component.find(".footer")).toHaveLength(1);
        expect(component.find(".footer__links-wrapper")).toHaveLength(1);
        expect(component.find(".footer__copyright")).toHaveLength(1);
    });

    it("doesn't render footer links if linksList is empty", () => {
        const emptyLinksList: FooterLinksType = [];
        const linksClass = "footer__links-wrapper";
        const component = shallow(<Footer linksList={emptyLinksList} />);

        expect(component.find(linksClass)).toHaveLength(0);
    });
});
