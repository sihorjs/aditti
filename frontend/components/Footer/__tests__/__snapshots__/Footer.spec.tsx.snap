// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders without crash 1`] = `ShallowWrapper {}`;
